elfpy/data/contract_interface.py

```diff
@@ -109,24 +112,104 @@ def fetch_and_decode_logs(web3: Web3, contract: Contract, tx_receipt: TxReceipt)
     return logs
 
 
-def fetch_transactions_for_block(
-    web3: Web3, contract: Contract, block_number: BlockNumber
-) -> list[dict[str, Any]] | None:
+def convert_fixedpoint(input_val: int | None) -> float | None:
+    """Given a scaled value int, converts it to an unscaled value in float, while dealing with Nones"""
+
+    # We cast to FixedPoint, then to floats to keep noise to a minimum
+    # This is assuming there's no loss of precision going from Fixedpoint to float
+    # Once this gets fed into postgres, postgres has fixed precision Numeric type
```

Comment:
Would it be better to pass the `fixedpoint.scaled_value` in directly, then? Casting to float can introduce noice (it is done via `float = fixedpoint.scaled_value / 1e18`, which will introduce noise that scales with the magnitude of the number).

I think this doesn't matter much in the post-processing phase, at least while our goal is only to make plots of macro trends. So feel free to leave it alone, just a thought.